silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -825,62 +800,34 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
     /// @dev Simulates a withdraw of `assets` from Morpho.
     /// @return The remaining assets to be withdrawn.
-    function _simulateWithdrawMorpho(uint256 assets) internal view returns (uint256) {
+    function _simulateWithdrawMorpho(uint256 assets) internal view virtual returns (uint256) {
         for (uint256 i; i < withdrawQueue.length; ++i) {
-            Id id = withdrawQueue[i];
-            MarketParams memory marketParams = _marketParams(id);
-
-            uint256 supplyShares = MORPHO.supplyShares(id, address(this));
-            (uint256 totalSupplyAssets, uint256 totalSupplyShares, uint256 totalBorrowAssets,) =
-                MORPHO.expectedMarketBalances(marketParams);
+            IERC4626 market = withdrawQueue[i];
 
             // The vault withdrawing from Morpho cannot fail because:
             // 1. oracle.price() is never called (the vault doesn't borrow)
             // 2. the amount is capped to the liquidity available on Morpho
             // 3. virtually accruing interest didn't fail
-            assets = assets.zeroFloorSub(
-                _withdrawable(
-                    marketParams,
-                    totalSupplyAssets,
-                    totalBorrowAssets,
-                    supplyShares.toAssetsDown(totalSupplyAssets, totalSupplyShares)
-                )
-            );
+            assets = UtilsLib.zeroFloorSub(assets, market.maxWithdraw(address(this)));
```

Comment:
I want to confirm that it is ok for us to use different functions to calculate assets to be withdrawn in the _withdrawMorpho fn and _simulateWithdrawMorpho fn.
```
uint256 toWithdraw = UtilsLib.min(market.maxWithdraw(address(this)), assets);
```
and 
```
assets = UtilsLib.zeroFloorSub(assets, market.maxWithdraw(address(this)));
```